File: apps/web-admin/src/types/lureType.ts
/**
 * 拟饵类型相关类型定义
 */

// ========== 分类枚举和配置 ==========

export type LureCategory = 'hard' | 'soft' | 'metal' | 'fly' | 'other'

export interface LureCategoryConfig {
  label: string
  icon: string
  color: string
  description: string
}

export const LURE_CATEGORY_CONFIG: Record<LureCategory, LureCategoryConfig> = {
  hard: {
    label: '硬饵',
    icon: '🎯',
    color: '#1890ff',
    description: '米诺、波趴、VIB等',
  },
  soft: {
    label: '软饵',
    icon: '🐛',
    color: '#52c41a',
    description: '卷尾蛆、T尾等',
  },
  metal: {
    label: '金属饵',
    icon: '✨',
    color: '#faad14',
    description: '亮片、铁板等',
  },
  fly: {
    label: '飞蝇',
    icon: '🦋',
    color: '#722ed1',
    description: '干蝇、湿蝇等',
  },
  other: {
    label: '其他',
    icon: '🎣',
    color: '#8c8c8c',
    description: '其他类型',
  },
}

// 分类选项列表
export const LURE_CATEGORY_OPTIONS = Object.entries(LURE_CATEGORY_CONFIG).map(
  ([value, config]) => ({
    value: value as LureCategory,
    icon: config.icon,
    color: config.color,
    label: `${config.icon} ${config.label}`,
  })
)

// ========== 拟饵类型接口 ==========

// 扩展信息接口（存储在 best_conditions JSON 中）
export interface LureTypeExtendedInfo {
  usage_tips?: string[]
  pros?: string[]
  cons?: string[]
  recommended_rod_power?: string[]
  recommended_line?: string
  retrieve_techniques?: string[]
  best_seasons?: string[]
  best_water_conditions?: string
  color_selection?: string
}

export interface LureType {
  lure_type_id: number
  name: string
  name_en?: string
  category: LureCategory
  description?: string
  action_description?: string
  best_conditions?: string
  target_species?: string
  typical_weight_min?: number
  typical_weight_max?: number
  image_url?: string
  created_at?: string
  updated_at?: string
}

export interface LureTypeListItem {
  lure_type_id: number
  name: string
  name_en?: string
  category: LureCategory
  target_species?: string
  typical_weight_min?: number
  typical_weight_max?: number
}

export interface LureTypeListResponse {
  items: LureTypeListItem[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

// ========== 请求类型 ==========

export interface LureTypeCreateRequest {
  name: string
  category: LureCategory
  description?: string
  action_description?: string
  best_conditions?: string
  target_species?: string
  typical_weight_min?: number
  typical_weight_max?: number
  image_url?: string
}

export interface LureTypeUpdateRequest {
  name?: string
  category?: LureCategory
  description?: string
  action_description?: string
  best_conditions?: string
  target_species?: string
  typical_weight_min?: number
  typical_weight_max?: number
  image_url?: string
}

export interface LureTypeListParams {
  page?: number
  page_size?: number
  category?: LureCategory
  keyword?: string
}

// ========== 分类统计 ==========

export interface CategoryStats {
  category: string
  count: number
  label: string
  icon: string
  color: string
}

export interface CategoryStatsResponse {
  categories: CategoryStats[]
  total: number
}

// ========== 初始化数据响应 ==========

export interface InitDataResponse {
  created_count: number
  message: string
}

// ========== LLM增强进度 ==========

export interface EnrichProgressItem {
  lure_type_id: number
  name: string
  status: 'pending' | 'enriching' | 'completed' | 'failed' | 'skipped'
  error?: string
  updated_at?: string
}

export interface EnrichProgressStats {
  total: number
  pending: number
  enriching: number
  completed: number
  failed: number
  skipped: number
}

export interface EnrichProgressResponse {
  is_running: boolean
  stats: EnrichProgressStats
  items: EnrichProgressItem[]
}

// ========== 网页采集进度 ==========

export interface CrawlProgressResponse {
  is_running: boolean
  status: 'idle' | 'crawling' | 'translating' | 'completed' | 'failed'
  message: string
  crawled_count: number
  translated_count: number
  total_count: number
  error?: string | null
}

File: apps/api/schemas/lure_type.py
"""
拟饵类型 Schema 定义

定义拟饵类型相关的请求/响应模型
"""

from datetime import datetime
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field


class LureCategoryEnum(str, Enum):
    """拟饵分类枚举"""

    HARD = "hard"  # 硬饵
    SOFT = "soft"  # 软饵
    METAL = "metal"  # 金属饵
    FLY = "fly"  # 飞蝇
    OTHER = "other"  # 其他


# ========== Create/Update Schemas ==========


class LureTypeCreate(BaseModel):
    """创建拟饵类型请求"""

    name: str = Field(..., min_length=1, max_length=100, description="拟饵名称")
    category: LureCategoryEnum = Field(..., description="拟饵分类")
    description: Optional[str] = Field(None, description="详细描述")
    action_description: Optional[str] = Field(None, description="动作描述")
    best_conditions: Optional[str] = Field(None, description="最佳使用条件")
    target_species: Optional[str] = Field(None, max_length=200, description="目标鱼种")
    typical_weight_min: Optional[float] = Field(
        None, ge=0, description="典型最小重量(克)"
    )
    typical_weight_max: Optional[float] = Field(
        None, ge=0, description="典型最大重量(克)"
    )
    image_url: Optional[str] = Field(None, max_length=500, description="图片URL")


class LureTypeUpdate(BaseModel):
    """更新拟饵类型请求"""

    name: Optional[str] = Field(None, min_length=1, max_length=100, description="拟饵名称")
    category: Optional[LureCategoryEnum] = Field(None, description="拟饵分类")
    description: Optional[str] = Field(None, description="详细描述")
    action_description: Optional[str] = Field(None, description="动作描述")
    best_conditions: Optional[str] = Field(None, description="最佳使用条件")
    target_species: Optional[str] = Field(None, max_length=200, description="目标鱼种")
    typical_weight_min: Optional[float] = Field(
        None, ge=0, description="典型最小重量(克)"
    )
    typical_weight_max: Optional[float] = Field(
        None, ge=0, description="典型最大重量(克)"
    )
    image_url: Optional[str] = Field(None, max_length=500, description="图片URL")


# ========== Response Schemas ==========


class LureTypeResponse(BaseModel):
    """拟饵类型详情响应"""

    lure_type_id: int
    name: str
    category: str
    description: Optional[str] = None
    action_description: Optional[str] = None
    best_conditions: Optional[str] = None
    target_species: Optional[str] = None
    typical_weight_min: Optional[float] = None
    typical_weight_max: Optional[float] = None
    image_url: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class LureTypeListItem(BaseModel):
    """拟饵类型列表项"""

    lure_type_id: int
    name: str
    category: str
    target_species: Optional[str] = None
    typical_weight_min: Optional[float] = None
    typical_weight_max: Optional[float] = None

    model_config = ConfigDict(from_attributes=True)


class LureTypeListResponse(BaseModel):
    """拟饵类型分页列表响应"""

    items: List[LureTypeListItem]
    total: int
    page: int
    page_size: int
    total_pages: int


# ========== Category Stats ==========


class CategoryStats(BaseModel):
    """分类统计"""

    category: str
    count: int
    label: str
    icon: str
    color: str


class CategoryStatsResponse(BaseModel):
    """分类统计响应"""

    categories: List[CategoryStats]
    total: int


# ========== Init Data Response ==========


class InitDataResponse(BaseModel):
    """初始化数据响应"""

    created_count: int
    message: str
